src/features/pastelDB/pastelDBThread.tsx

```diff
@@ -0,0 +1,227 @@
+  const result = await getMiningInfo(props.rpcConfig)
+  insertMiningInfoToDB(props.pastelDB, result)
+}
+
+const fetchBlock = async (props: fetchFuncConfig) => {
+  const result = await getBlockByHash(props.rpcConfig)
+  insertBlockInfoToDB(props.pastelDB, result)
+}
+
+const fetchRawtransaction = async (props: fetchFuncConfig) => {
+  const listSinceBlock = await getListSinceBlock(props.rpcConfig)
+  for (let i = 0; i < listSinceBlock.transactions.length; i++) {
+    const transaction: TSinceblockTransaction = listSinceBlock.transactions[i]
+    const result = await getRawTransaction(transaction.txid, props.rpcConfig)
+    insertRawtransaction(props.pastelDB, result)
+  }
+}
+
+const fetchTransaction = async (props: fetchFuncConfig) => {
+  const listSinceBlock = await getListSinceBlock(props.rpcConfig)
+  for (let i = 0; i < listSinceBlock.transactions.length; i++) {
+    const transaction: TSinceblockTransaction = listSinceBlock.transactions[i]
+    const result = await getWalletTransaction(transaction.txid, props.rpcConfig)
+    insertTransaction(props.pastelDB, result)
+  }
+}
+
+const fetchTxoutsetinfo = async (props: fetchFuncConfig) => {
+  const result = await getRpcData(props.rpcConfig)
+  insertTxoutsetinfo(props.pastelDB, result)
+}
+
+const fetchChaintips = async (props: fetchFuncConfig) => {
+  const chaintips = await getChaintips(props.rpcConfig)
+  for (let i = 0; i < chaintips.length; i++) {
+    insertChaintips(props.pastelDB, chaintips[i])
+  }
+}
+
+const fetchBlocksubsidy = async (props: fetchFuncConfig) => {
+  const result = await getBlockSubSidy(props.rpcConfig)
+  insertBlocksubsidy(props.pastelDB, result)
+}
+
+const fetchWalletinfo = async (props: fetchFuncConfig) => {
+  const result = await getWalletInfo(props.rpcConfig)
+  insertWalletinfo(props.pastelDB, result)
+}
+
+const fetchListTransactions = async (props: fetchFuncConfig) => {
+  const transactions = await listTransactions(props.rpcConfig)
+  for (let i = 0; i < transactions.length; i++) {
+    insertListTransactions(props.pastelDB, transactions[i])
+  }
+}
+
+const fetchListunspent = async (props: fetchFuncConfig) => {
+  const unspentlist = await listUnspent(props.rpcConfig)
+  for (let i = 0; i < unspentlist.length; i++) {
+    insertListunspent(props.pastelDB, unspentlist[i])
+  }
+}
+
+const fetchTotalbalance = async (props: fetchFuncConfig) => {
+  const result = await getTotalBalance(props.rpcConfig)
+  insertTotalbalance(props.pastelDB, result)
+}
+
+const fetchListaddresses = async (props: fetchFuncConfig) => {
+  const addresslist = await listAddresses(props.rpcConfig)
+  for (let i = 0; i < addresslist.length; i++) {
+    insertListaddresses(props.pastelDB, addresslist[i])
+  }
+}
+
+export const PastelDBThread = (
+  props: PastelDBThreadProps,
+): React.ReactElement => {
+  const { rpcConfig } = props
+  const [isStarted, setStarted] = useState(false)
+
+  useEffect(() => {
+    const pastelDB = remote.getGlobal('pastelDB')
```

Comment:
I think we fixed this, so it's not global anymore (there was a `database` object)